packages/web-app-files/tests/unit/components/Search/List.spec.js

```diff
@@ -0,0 +1,189 @@
+import { mount, createLocalVue } from '@vue/test-utils'
+import Vuex from 'vuex'
+import DesignSystem from 'owncloud-design-system'
+import GetTextPlugin from 'vue-gettext'
+
+import List from '@files/src/components/Search/List.vue'
+
+const localVue = createLocalVue()
+localVue.use(Vuex)
+localVue.use(DesignSystem)
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+const stubs = {
+  'no-content-message': false,
+  'oc-table-files': false,
+  pagination: true,
+  'list-info': true
+}
+
+const selectors = {
+  noContentMessage: '.files-empty',
+  filesTable: '.files-table',
+  pagination: 'pagination-stub',
+  listInfo: 'list-info-stub'
+}
+
+const files = [
+  {
+    path: 'lorem.txt',
+    size: 100
+  },
+  {
+    path: 'lorem.pdf',
+    size: 50
+  }
+]
+
+describe('List component', () => {
+  afterAll(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('when search term is not provided', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapper()
+    })
+
+    it('should show text "No search term entered"', () => {
+      const noContentMessage = wrapper.find(selectors.noContentMessage)
+      const filesTable = wrapper.find(selectors.filesTable)
+
+      expect(filesTable.exists()).toBeFalsy()
+      expect(noContentMessage.exists()).toBeTruthy()
+      expect(noContentMessage.text()).toEqual('No search term entered')
+    })
+    it('should not show files table', () => {
+      const filesTable = wrapper.find(selectors.filesTable)
+
+      expect(filesTable.exists()).toBeFalsy()
+    })
+  })
+
+  describe('when resource is not found', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapper('epsum')
+    })
+
+    it('should show text "No resource found"', () => {
+      const noContentMessage = wrapper.find(selectors.noContentMessage)
+      const filesTable = wrapper.find(selectors.filesTable)
+
+      expect(filesTable.exists()).toBeFalsy()
+      expect(noContentMessage.exists()).toBeTruthy()
+      expect(noContentMessage.text()).toEqual('No resource found')
+    })
+    it('should not show files table', () => {
+      const filesTable = wrapper.find(selectors.filesTable)
+
+      expect(filesTable.exists()).toBeFalsy()
```

Comment:
already tested in the scenario just before this
maybe just putting it here will be nice